src/main/java/org/generationcp/middleware/service/impl/crop/CropGenotypingParameterServiceImpl.java

```diff
@@ -1,66 +1,69 @@
 package org.generationcp.middleware.service.impl.crop;
 
+import org.apache.commons.lang3.StringUtils;
 import org.generationcp.middleware.hibernate.HibernateSessionProvider;
-import org.generationcp.middleware.manager.WorkbenchDaoFactory;
-import org.generationcp.middleware.pojos.workbench.CropGenotypingParameter;
-import org.generationcp.middleware.service.api.crop.CropGenotypingParameterMapper;
+import org.generationcp.middleware.manager.DaoFactory;
+import org.generationcp.middleware.pojos.CropParameter;
 import org.generationcp.middleware.service.api.crop.CropGenotypingParameterService;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 
 @Service
 @Transactional
 public class CropGenotypingParameterServiceImpl implements CropGenotypingParameterService {
 
-	private final WorkbenchDaoFactory workbenchDaoFactory;
+	private final DaoFactory daoFactory;
```

Comment:
I think it's better to just use CropParameterResource to get the parameters related to Germplasm Details > Genotyping tab. And just remove this and related classes altogether (e.g. CropGenotypingParameterResource, etc).